src/vcpkg/archives.cpp

```diff
@@ -356,4 +356,21 @@ namespace vcpkg
 
         return filtered_results;
     }
+
+    ExpectedL<Unit> split_archive(Filesystem& fs, const Path& path, int64_t file_size, int64_t chunk_size)
+    {
+        auto file = fs.open_for_read(path, VCPKG_LINE_INFO);
+
+        std::vector<unsigned char> buffer(chunk_size);
```

Comment:
If we come anywhere close to needing an int64_t then those chunks are too big and we shouldn't be trying to allocate them all at once.

This also isn't doing the right retries in the case of partial reads or writes. For example, it moves forward by chunk_size but there's no guarantee that bytes_read == chunk_size.

I think there is already code in the copy_file bits in files.cpp you could steal as a starting point.

Reply:
> If we come anywhere close to needing an int64_t then those chunks are too big and we shouldn't be trying to allocate them all at once.

The intention is that whoever calls the function is responsible to pass a reasonable value. What would you suggest I do? There are many ways to address your concerns.

Reply:
>The intention is that whoever calls the function is responsible to pass a reasonable value. 

I don't think it is possible for callers to do that. This is a file size. Files routinely do not fit in RAM. If it needs to copy the whole file into RAM, then it should be appropriately limited by accepting a `size_t` or `ptrdiff_t` rather than an `int64_t`. That way callers get appropriate warnings and have to deal with the condition that the number they showed up with does not fit.